### PR TITLE
Add DynamicReader, DynamicWriter tests [Core DLL]

### PR DESCRIPTION
**Pull request by aeulitz**
DynamicReader and DynamicWriter components are already being built into the Win32 DLL; this change adds tests for them. As DynamicReader and DynamicWriter can normally not be activated, this change adds a helper component ('Microsoft.Internal.TestController') that allows us to create these components for test purposes. Once the ABI has been fully built up, we should be able to test these components in different ways. At this point, the Microsoft.Internal.TestController component can be removed again.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5455)

### CODE REVIEW

File: vnext/Desktop.ABITests/DynamicReaderWriterTests.cpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    writer.WriteInt64(184827);
+
+    writer.WriteObjectEnd();
+
+    IJSValueReader reader = Microsoft::Internal::TestController::CreateDynamicReader(writer);
+    TestCheckEqual(JSValueType::Object, reader.ValueType());
+
+    TestCheck(reader.GetNextObjectProperty(namePropertyName));
+    TestCheckEqual(JSValueType::String, reader.ValueType());
+    TestCheckEqual(L"Washington", reader.GetString());
```
Comment (dannyvv) [inline vnext/Desktop.ABITests/DynamicReaderWriterTests.cpp:137]: maybe a good testcase would be to read the properties out of order as well.

File: vnext/Desktop.ABITests/DynamicReaderWriterTests.cpp [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    writer.WritePropertyName(areaPropertyName);
+    writer.WriteInt64(184827);
+
+    writer.WriteObjectEnd();
+
+    IJSValueReader reader = Microsoft::Internal::TestController::CreateDynamicReader(writer);
+    TestCheckEqual(JSValueType::Object, reader.ValueType());
+
+    TestCheck(reader.GetNextObjectProperty(namePropertyName));
```
Comment (dannyvv) [inline vnext/Desktop.ABITests/DynamicReaderWriterTests.cpp:135]: >namePropertyName [](start = 43, length = 16)

my understanding is that this is an out value to the properName.
therefore we should probablye declare an `hstring currentProperty` and pass that here and have a `TestCheckEqual(namePropertyName, currentProperty)`

Comment (aeulitz) [reply vnext/Desktop.ABITests/DynamicReaderWriterTests.cpp:135]: That would make sense, let me try.

File: vnext/Desktop.ABITests/DynamicReaderWriterTests.cpp [hunk 3]
```diff
@@ -68,76 +68,80 @@ TEST_CLASS (DynamicReaderWriterTests) {
...
 
-    writer.WritePropertyName(namePropertyName);
+    writer.WritePropertyName(L"Name");
     writer.WriteString(L"Bob");
 
-    writer.WritePropertyName(agePropertyName);
+    writer.WritePropertyName(L"Age");
     writer.WriteInt64(32);
 
-    writer.WritePropertyName(marriedPropertyName);
```
Comment (dannyvv) [inline vnext/Desktop.ABITests/DynamicReaderWriterTests.cpp]: Nit: Maybe a more general boolean property: "hasBeenToSpace"

File: vnext/Desktop/ABI/TestController.idl [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import "IJSValueReader.idl";
```
Comment (vmoroz) [inline vnext/Desktop/ABI/TestController.idl:1]: >import [](start = 0, length = 6)

Please add the file header.

File: vnext/Desktop/ABI/TestController.h [hunk 5]
```diff
@@ -0,0 +1,16 @@
+#pragma once
```
Comment (vmoroz) [inline vnext/Desktop/ABI/TestController.h:1]: >pragma [](start = 1, length = 6)

Please add file headers to all new files.

**Pull request merged**